xdsl/backend/csl/print_csl.py

```diff
@@ -0,0 +1,170 @@
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from typing import IO
+
+from xdsl.dialects import arith, func
+from xdsl.dialects.builtin import (
+    Float16Type,
+    Float32Type,
+    FloatAttr,
+    IndexType,
+    IntAttr,
+    IntegerAttr,
+    IntegerType,
+    ModuleOp,
+    Signedness,
+    SignednessAttr,
+)
+from xdsl.ir import Attribute, Block, SSAValue
+
+
+@dataclass
+class CslPrintContext:
+    output: IO[str]
+
+    variables: dict[SSAValue, str] = field(default_factory=dict)
+
+    _counter: int = field(default=0)
+
+    _prefix: str = field(default="")
+
+    def print(self, text: str, prefix: str = ""):
+        """
+        Print `text` line by line, prefixed by self._prefix and prefix.
+        """
+        for l in text.split("\n"):
+            print(self._prefix + prefix + l, file=self.output)
+
+    def _get_variable_name_for(self, val: SSAValue) -> str:
+        """
+        Get an assigned variable name for a given SSA Value
+        """
+        if val in self.variables:
+            return self.variables[val]
+
+        taken_names = set(self.variables.values())
+        if val.name_hint is not None and val.name_hint not in taken_names:
+            name = val.name_hint
+        else:
+            prefix = "v" if val.name_hint is None else val.name_hint
+
+            name = f"{prefix}{self._counter}"
+            self._counter += 1
+
+            while name in taken_names:
+                name = f"{prefix}{self._counter}"
+                self._counter += 1
+
+        self.variables[val] = name
+        return name
+
+    def mlir_type_to_csl_type(self, type_attr: Attribute) -> str:
+        """
+        Convert an MLR type to a csl type. CSL supports a very limited set of types:
+
+        - integer types: i16, u16, i32, u32
+        - float types: f16, f32
+        - pointers: [*]f32
+        - arrays: [64]f32
+
+        This method does not yet support all the types and will be expanded as needed later.
+        """
+        match type_attr:
+            case Float16Type():
+                return "f16"
+            case Float32Type():
+                return "f32"
+            case (
+                IndexType()
+            ):  # TODO: figure out what the equivalent of intpr_t is for csl!
+                return "i64"
+            case IntegerType(
+                width=IntAttr(data=width),
+                signedness=SignednessAttr(data=Signedness.UNSIGNED),
+            ):
+                return f"u{width}"
+            case IntegerType(width=IntAttr(data=width)):
+                return f"i{width}"
+            case unkn:
+                return f"<!unknown type {unkn}>"
```

Comment:
nit:
```suggestion
            case _:
                return f"<!unknown type {type_attr}>"
```
`unkn` threw me off a bit

Reply:
similarly below